support-services/src/main/scala/com/gu/support/redemption/RedemptionCode.scala

```diff
@@ -0,0 +1,13 @@
+package com.gu.support.redemption
+
+object RedemptionCode {
+  def apply(codeAsString: String): Either[String, RedemptionCode] = {
+    // make sure no one can inject anything bad
+    val validChars = (('A' to 'Z') ++ ('0' to '9') ++ Seq('-')).toSet
+    if (codeAsString.forall(validChars.contains))
+      Right(new RedemptionCode(codeAsString))
+    else
+      Left(s"redemption code must only include ${validChars.mkString("")}")
+  }
+}
+case class RedemptionCode private (codeAsString: String) extends AnyVal
```

Comment:
I would say `value` is more consistent with other classes in this codebase than `codeAsString`

Reply:
I used to do the same until recently, but then I realised if we call them all `value`, then it's going the same way as the ubiquitous `toString` that will compile regardless, but won't always do what you expect.  I don't feel as strongly as for toString though so happy to change it.